### PR TITLE
rgw: fix vector index warning

### PR DESCRIPTION
**Pull request by changchengx**
src/rgw/rgw_etag_verifier.cc: In member function ‘virtual int RGWPutObj_ETagVerifier_MPU::process(ceph::bufferlist&&, uint64_t)’:
src/rgw/rgw_etag_verifier.cc:58:23: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<long unsigned int>::size_type
   58 |   if (next_part_index == part_ofs.size()) {
      |       ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
src/rgw/rgw_etag_verifier.cc:76:25: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<long unsigned int>::size_type
   76 |     if (next_part_index == part_ofs.size())
      |         ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~

Signed-off-by: Changcheng Liu <changcheng.liu@aliyun.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by changchengx** [comment]
test this please

**Comment by changchengx** [comment]
@tchaikov Please help review

**Comment by changchengx** [comment]
@tchaikov

File: src/rgw/rgw_etag_verifier.h [hunk 1]
```diff
@@ -54,7 +54,7 @@ class RGWPutObj_ETagVerifier_Atomic : public RGWPutObj_ETagVerifier
 class RGWPutObj_ETagVerifier_MPU : public RGWPutObj_ETagVerifier
 {
   std::vector<uint64_t> part_ofs;
-  int cur_part_index{0}, next_part_index{1};
+  decltype(part_ofs)::size_type cur_part_index{0}, next_part_index{1};
```
Comment (tchaikov) [inline src/rgw/rgw_etag_verifier.h:57]: i'd just go with unsigned instead worrying about the type of `part_ofs`.

**Comment by changchengx** [comment]
jenkins, test this please

**Comment by tchaikov** [comment]
jenkins test make check

**Review by tchaikov — APPROVED** [review]

**Comment by changchengx** [comment]
@tchaikov ping

**Comment by tchaikov** [comment]
> @tchaikov ping

i am not qualified to check the rgw test results. and in my last rgw run, there were quite a few test failures in my rgw run, so i'd prefer deferring the test to rgw team.

**Review by cbodley — APPROVED** [review]

**Comment by cbodley** [comment]
@tchaikov thanks for help with review!

**Pull request merged**